nevergrad/optimization/optimizerlib.py

```diff
@@ -1815,6 +1817,171 @@ def __init__(
 )
 
 
+@registry.register
+class PCABO(base.Optimizer):
+    """Principle Component Analysis (PCA) Bayesian Optimization for dimensionality reduction in BO
+
+    References
+
+    [RaponiWB+20]
+        Raponi, Elena, Hao Wang, Mariusz Bujny, Simonetta Boria, and Carola Doerr.
+        "High dimensional bayesian optimization assisted by principal component analysis."
+        In International Conference on Parallel Problem Solving from Nature, pp. 169-183.
+        Springer, Cham, 2020.
+
+    """
+
+    no_parallelization = True
+    recast = True
+
+    def __init__(
+        self,
+        parametrization: IntOrParameter,
+        budget: tp.Optional[int] = None,
+        num_workers: int = 1,
+        *,
+        init_budget: tp.Optional[int] = None,
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+
+        from bayes_optim.extension import PCABO as PcaBO
+        from bayes_optim.extension import RealSpace
+
+        lb, ub = 1e-7 - np.pi / 2, np.pi / 2 - 1e-7
+        space = RealSpace([lb, ub]) * self.dimension
+        self.buffer: List[float] = []
+        self.newX: List[float] = []
+        self.loss: List[float] = []
+
+        self._pcabo = PcaBO(
+            search_space=space,
+            obj_fun=None,  # Assuming that this is not used :-)
+            DoE_size=init_budget if init_budget is not None else 5,
+            max_FEs=budget,
+            verbose=True,
+            n_point=1,  # We start with a sequential procedure, maybe we'll extend in a second moment
+            n_components=0.95,
+            acquisition_optimization={"optimizer": "BFGS"},
+        )
+
+    def _internal_ask_candidate(self) -> p.Parameter:
+        if not self.buffer:
+            self.newX = []
+            self.loss = []
+            candidate = self._pcabo.ask()
+            if not isinstance(candidate, list):
+                candidate = candidate.tolist()
+            self.buffer = candidate
+        x_probe = self.buffer.pop()
+        print("X_PROBE = ", x_probe)
+        print("BUFFER = ", self.buffer)
```

Comment:
please remove the prints when everything is working

Reply:
Ok